packages/plugin-ext/src/hosted/node/hosted-plugin-process.ts

```diff
@@ -41,6 +41,8 @@ export class HostedPluginProcess implements ServerPluginRunner {
     @inject(HostedPluginCliContribution)
     protected readonly cli: HostedPluginCliContribution;
 
+    @multiInject(PluginHostEnvironmentVariable) pluginHostEnvironmentVariables: PluginHostEnvironmentVariable[];
```

Comment:
There are several issues with mulit inject:
- contribution points are not discoverable 
- it produces new instance for each inject
- the real type is `PluginHostEnvironmentVariable[] | undefined`, i.e.it won't work if there are non client contributing

There is [ContributionProvider](https://github.com/theia-ide/theia/blob/636533952fd4b42cfc834e346ec3402f0c5db343/packages/core/src/common/contribution-provider.ts#L69) which fixes all these flaws.

Reply:
ok

Reply:
fixed